web/source/osk/visualKeyboard.ts

```diff
@@ -703,45 +697,34 @@ namespace com.keyman.osk {
         return;
       }
 
-      // Clear previous key highlighting
-      if(!this.subkeyPopup && key0 && key1 && key1 !== key0) {
-        this.highlightKey(key0,false);
-      }
-
-      // Do not move over keys if device popup visible
-      if(this.subkeyDelegator) {
-        if(key1 == null) {
-          if(key0) {
-            this.highlightKey(key0,false);
-          }
-          this.keyPending=null;
-          this.touchPending=null;
-        } else {
-          // Re-apply highlighting to the key if it's the current popup's base key.
-          if(this.subkeyDelegator && key1 == this.subkeyDelegator.baseKey) {
-            if(!key1.classList.contains('kmw-key-touched')) {
-              this.highlightKey(key1,true);
-            }
-            this.keyPending = key1;
-            this.touchPending = e.touches[0];
-          } else {
-            if(key0) {
-              this.highlightKey(key0,false);
-            }
-            this.keyPending = null;
-            this.touchPending = null;
-          }
```

Comment:
This is the _other_ part of that noted block from SubkeyDelegator; it's what helped me identify the need for that bug fix.  Why highlight the key if it won't actually output?

Also, note:  the initial `if` and the inner `else` were literally identical.